proxy/hdrs/URL.cc

```diff
@@ -1759,14 +1759,16 @@ url_MD5_get_general(const URLImpl *url, CryptoContext &ctx, CryptoHash &hash, ca
   if (p != buffer) {
     ctx.update(buffer, p - buffer);
   }
-
-  port = url_canonicalize_port(url->m_url_type, url->m_port);
+  int buffer_len = static_cast<int>(p - buffer);
+  port           = url_canonicalize_port(url->m_url_type, url->m_port);
 
   ctx.update(&port, sizeof(port));
   if (generation != -1) {
     ctx.update(&generation, sizeof(generation));
+    Debug("url", "Final url string for hashing %.*s%d%d", buffer_len, buffer, port, static_cast<int>(generation));
```

Comment:
Hmmm, "url" seems like an odd tag here. I think it should be "http_<something". url  is simply much to generic, a diagnostic on "-T url" would also include all url_rewrite Debug tags for example (yes, I know you can anchor the tag, like ^url$, but still.).

Reply:
Yes, we should err towards more verbose tags. I would recommend `url-cache` at a minimum. Also, "hashing" is a bit vague as well, it should be "cache hash key". We do a lot of different hashing these days and the whole point of this is better clarity.

Reply:
well, do we use this function only caching purposes? Since this is a general function to calculate hash of the URL string. If yes, then <url-cache>  and <cache hash key> work